### ISSUE TITLE
viewer_examples/plugins/watershed_demo.py not updated for Qt5

### ISSUE THREAD

**Issue by jmetz** [scikit-image/scikit-image#2745]
## Description

Error presumably related to this example not having been updated for Qt5:

```
$ python viewer_examples/plugins/watershed_demo.py           
Watershed plugin
----------------
Use mouse to paint each region with a different label.
Press OK to display segmented image.
Traceback (most recent call last):
  File "viewer_examples/plugins/watershed_demo.py", line 39, in <module>
    plugin += OKCancelButtons()
  File "skimage/viewer/widgets/history.py", line 25, in __init__
    self.ok = QtGui.QPushButton('OK')
AttributeError: module 'PyQt5.QtGui' has no attribute 'QPushButton'
```

## Way to reproduce

- Simple type `python viewer_examples/plugins/watershed_demo.py` from repo directory
- System: `Linux 4.9.0-3-amd64 #1 SMP Debian 4.9.30-2+deb9u1 (2017-06-18) x86_64 GNU/Linux`
- Python: `Python 3.6.1`
- `skimage.__version__` : `0.14dev`

**Comment by soupault** [comment]
:+1: to update. I think, Qt5 has become a standard since recently.

**Comment by jmetz** [comment]
Ah... hadn't realised this wasn't already the de-facto standard for `skimage`, as I also tested the `viewer_examples/plugins/canny.py` and the `viewer_examples/plugins/lineprofile.py` and they worked fine. 

A little confused now how they can be working fine and not the `watershed.py` example if Qt5 isn't already the standard for `skimage`?

**Comment by jmetz** [comment]
@soupault  perhaps this should be renamed 

> skimage/viewer/widgets/history.py doesn't work with Qt5

as the bug seems to just be a symptom of that, right?

**Comment by soupault** [comment]
@jmetz Sorry if I was unclear. I meant that Qt5, to the best of my knowledge, is a standard across the globe for the last 1-2 years. Before that, Qt5 hadn't been considered as a favorite.

In `skimage`, currently, we state that the dependency is PyQt4 (https://github.com/scikit-image/scikit-image/blob/4d4c79966d28a3b80a04b7462bc3cd28c362c269/DEPENDS.txt), and I'd be glad to support the update to PyQt5.

**Comment by jmetz** [comment]
Thanks for clarifying. I guess lot's of systems do still use Qt4, and Matplotlib has had the option of either for quite some time... do you think we could offer the same choices for `skimage`? 

This could mean having an internal ui submodule, which all other `skimage` components use, and in that ui module, qt4, qt5 (etc) are switched between depending on what's available / been chosen by the user.

**Comment by stefanv** [comment]
`matplotlib` has a compatibility layer for importing; can we re-use theirs?

**Comment by jmetz** [comment]
Possibly, though that would scrap #2710 

From what I can tell, almost all of the cross Qt4-Qt5 functionality is provided by using 

```python
QtWidgets = QtGui
```
in https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/qt_compat.py#L220, as that was one of the core differences anyway. 
Any additional minor changes are then handled in the respective `backend_qtX.py` files.

**Comment by stefanv** [comment]
Great, that simplifies things; thanks for investigating.

**Comment by jni** [comment]
This should also be completely fixed in #3210

**Comment by jmetz** [comment]
@soupault & @jni - this now loads the interface for me but fails on click: 

```shell
skimage [master●●] % python3.6 viewer_examples/plugins/watershed_demo.py
Watershed plugin
----------------
Use mouse to paint each region with a different label.
Press OK to display segmented image.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/matplotlib/cbook/__init__.py", line 388, in process
    proxy(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/matplotlib/cbook/__init__.py", line 228, in __call__
    return mtd(*args, **kwargs)
  File "/mirrored/home/jeremy/Documents/programming/python/others/scikit-image/skimage/viewer/utils/canvas.py", line 75, in on_mouse_press
    self.active_tool.on_mouse_press(event)
  File "/mirrored/home/jeremy/Documents/programming/python/others/scikit-image/skimage/viewer/canvastools/painttool.py", line 149, in on_mouse_press
    self.update_overlay(event.xdata, event.ydata)
  File "/mirrored/home/jeremy/Documents/programming/python/others/scikit-image/skimage/viewer/canvastools/painttool.py", line 172, in update_overlay
    overlay[self.window.at(y, x)] = self.label
TypeError: slice indices must be integers or None or have an __index__ method
```

i.e. presumably the issue is that `self.window.at(y,x)` is not an object that can be used for slicing.

That said, this seems to now be an issue for `skimage.viewer.canvastools.painttool`.

**Comment by jni** [comment]
@jmetz please see #3210

**Comment by jni** [comment]
(It’s not merged yet)

**Comment by jmetz** [comment]
@jni - ah right, sorry just realised you fixed that there. :+1:

**Comment by soupault** [comment]
Closed via #3210
